actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,34 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :==, :each, :empty?, to: :variants
+
+        def initialize(variants = [])
+          @variants = variants
+        end
+
+        def any?(*candidates)
+          (variants & candidates).any?
+        end
+
+        def to_ary
+          variants
+        end
+
+        protected
+          attr_reader :variants
```

Comment:
I still wonder why this can't be public since you can access it via `to_ary` also.